test/upgrade/prober/wathola/event/services.go

```diff
@@ -91,7 +101,7 @@ func (f *finishedStore) RegisterFinished(finished *Finished) {
 	time.Sleep(d)
 	receivedEvents := f.steps.Count()
 	if receivedEvents != finished.Count {
-		f.errors.throw("expecting to have %v unique events received, "+
+		f.errors.throwUnexpected("expecting to have %v unique events received, "+
```

Comment:
Ok. You are using unexpected, but still it will be reported as error. And each missing events will be reported. So, in case of missing events we would always see: `n+1` number of errors if we lose `n` events.

But, that's not related to your PR, so we can follow up that in another PR.